fit1d.py

```diff
@@ -0,0 +1,328 @@
+        sigma clipping
+
+    grow : float
+        Used to calculate how many elements adjacent to reject pixels are also rejected.
+        Not currently implemented, but included to duplicate the IRAF fitting parameter.
+
+    Returns
+    -------
+
+    fit : scipy.interpolate.fitpack2.LSQUnivariateSpline object
+        The result of the fit.  It can be used to calculate the fitted
+        values for the input x values:
+
+        fitted_values = fit(x)
+
+    """
+
+    nx = len(x)
+    ny = len(y)
+    if nx != ny:
+        print("X and Y vectors have different lengths, aborting fit1d")
+        return None
+    if weights is not None:
+        nw = len(weights)
+        if nw != nx:
+            print("Weight vector has different length to data vectors, aborting fit1d")
+            print("Use weight=None to assign equal weights")
+            return None
+    else:
+        weights = np.ones(nx)
+    xfit, yfit, wfit = wtrebin(x, y, weights, naverage)
+    fitted = fit_with_rejection(xfit, yfit, wfit, function, order, low_reject, high_reject,
+                                niterate, grow)
+    return fitted
+
+def get_knots(x, number_of_knots):
+    """Calculate equally-spaced knots for an array, given the array and the number of knots
+    to calculate.  The number of knots is 1 fewer than the order of the fit.
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        The input array for which knots are to be calculated
+
+    number_of_knots : int
+        The number of knots to calculate
+
+    Returns
+    -------
+
+    knots : 1-d ndarray
+        The array of equally-spaced knots
+
+    """
+
+    interval = x[-1] - x[0]
+    #
+    # If there are n knots, we are dividing the interval into (n+1) regions
+    #
+    subinterval = interval/float(number_of_knots + 1)
+    knots = x[0] + subinterval*np.arange(number_of_knots + 1)
+    #
+    # First knot is at x[0], we don't need that one
+    return knots[1:]
+
+def fit_once(x, y, weights, function, order):
+    """Do a single fit of function to (x, y) data.
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        Input array of x-values
+
+    y : 1-d array
+        Input array of y-values
+
+    weights : 1-d array
+        Input array of weights
+
+    function : str
+        Function to be fitted.  Currently only "spline1" and "spline3" are supported
+
+    order : int
+        Order of function to be fitted.  Since only splines are currently supported,
+        the order is the same as the number of equal reqions the data domain is split into,
+        and 1 more than the number of equally-spaced knots.
+
+    Returns
+    -------
+
+    fit : scipy.interpolate.fitpack2.LSQUnivariateSpline object
+        This can be used to calculate the fitted values for the input x values:
+
+        fitted_values = fit(x)
+
+    """
+
+    nx = len(x)
+    if function[:6] == "spline":
+        #
+        # Number of knots is 1 fewer than the order
```

Comment:
I would have expected the number of knots to be order + 1.  The reason it's order - 1 is that the scipy fitting routine requires that the knots be internal, i.e. there is no knot at either endpoint.